app/api/ada/adaTransactions/adaNewTransactions.js

```diff
@@ -46,107 +28,149 @@ const protocolMagic = CONFIG.network.protocolMagic;
 /** Calculate the transaction fee without actually sending the transaction */
 export async function getAdaTransactionFee(
   receiver: string,
-  amount: string
+  amount: string,
+  changeAdaAddr: ?AdaAddress,
+  possibleInputAddresses: Array<AdaAddress>,
+  addressesToUtxos: Array<string> => Promise<Array<UTXO>>
 ): Promise<AdaFeeEstimateResponse> {
-  const { txBuilder } = await newAdaUnsignedTx(receiver, amount);
-  const fee = txBuilder.estimate_fee(RustModule.Wallet.LinearFeeAlgorithm.default());
-  return { fee: fee.to_str() };
+  const { txBuilder } = await newAdaUnsignedTx(
+    receiver,
+    amount,
+    changeAdaAddr,
+    possibleInputAddresses,
+    addressesToUtxos
+  );
+  const fee = txBuilder.get_balance_without_fees().value();
```

Comment:
this reads pretty odd. Why is using `get_balance_without_fees()` rather than `estimate_fee`?

Reply:
I added more comments to explain what this means